internal/services/recommender.go

```diff
@@ -75,6 +75,19 @@ func ProcessEvent(msg *kafka.Message) {
 				} else {
 					log.Infof("Recommendation saved for experiment - %s and end_interval - %s", kafkaMsg.Experiment_name, recommendationSet.MonitoringEndTime)
 				}
+
+				// Create entry into HistoricalRecommendationSet table.
+				historicalRecommendationSet := model.HistoricalRecommendationSet{
+					WorkloadID:          kafkaMsg.WorkloadID,
+					ContainerName:       container.Container_name,
+					MonitoringStartTime: v.Duration_based.Short_term.Monitoring_start_time,
+					MonitoringEndTime:   v.Duration_based.Short_term.Monitoring_end_time,
+					Recommendations:     marshalData,
+				}
+				if err := historicalRecommendationSet.CreateHistoricalRecommendationSet(); err != nil {
```

Comment:
Here, it creates entry in both tables but not deleting an existing entry from recommendation_set table. Is it intentional?

Reply:
In the PR - [here](https://github.com/RedHatInsights/ros-ocp-backend/pull/104/files#diff-fe0fff74011513527c16ac9434707cd2add9039b21e9967b12f487f42832be49R86) we have db clause OnConflict. so if we try to add new entry in the table and if table already has entry for that workload then it just updates the recommendation in that entry. Hence no need of deletion.